frontend/src/pages/financialassistant/FinancialAssistant.tsx

```diff
@@ -0,0 +1,150 @@
+import React, { useEffect, useState } from "react";
+import { useAppContext } from "../../providers/AppProviders";
+import { upgradeSubscription, removeFinancialAssistant } from "../../api";
+import {
+    Spinner,
+    PrimaryButton,
+    DefaultButton,
+    Dialog,
+    DialogType,
+    DialogFooter,
+    MessageBar,
+    MessageBarType,
+} from "@fluentui/react";
+import styles from "./FinancialAssistant.module.css";
+
+const FinancialAssistant = () => {
+    const { user, organization } = useAppContext();
+    const [subscriptionStatus, setSubscriptionStatus] = useState<boolean | null>(null);
+    const [loading, setLoading] = useState(true);
+    const [showSubscribeDialog, setShowSubscribeDialog] = useState(false);
+    const [showUnsubscribeDialog, setShowUnsubscribeDialog] = useState(false);
+    const [error, setError] = useState<string | null>(null);
+
+    useEffect(() => {
+        if (user?.role !== "admin") {
+            return;
+        }
+
+        const fetchStatus = async () => {
+            setLoading(true);
+            try {
+                const response = await fetch(`/api/subscription/${organization?.subscriptionId}/financialAssistant/status`, {
```

Comment:
@ramirezmorac2, please implement this request as it is currently done in the api.tsx files.